plugins/Synchro/SynchroSynth.h

```diff
@@ -0,0 +1,154 @@
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ */
+
+#ifndef SYNCHROSYNTH_H
+#define SYNCHROSYNTH_H
+
+#include "Graph.h" //Required for the waveform view
+#include "Instrument.h" //Required for the synth
+#include "InstrumentView.h" //Required for instrument GUI
+#include "Knob.h" //Required to use knobs
+#include "NotePlayHandle.h" //Required for audio rendering
+
+constexpr char SYNCHRO_VERSION [] = "0.6";
+constexpr float SYNCHRO_VOLUME_CONST = 0.15f; //Prevents clipping
+constexpr int SYNCHRO_OVERSAMPLE = 4; //Anti-aliasing samples
+constexpr int SYNCHRO_PM_CONST = 20; //Strength of the phase modulation
+constexpr int SYNCHRO_GRAPH_SAMPLES = 48; //Resolution of the waveform view
+constexpr float MAGIC_HARMONICS[2][2] = {{32, 0.5}, {38, 0.025}}; //Yoioioi
+
+//Used to pass the oscillator settings from the GUI to the synth itself
+struct SynchroOscillatorSettings
+{
+	float Detune; //Detune in octaves
+	float Drive; //Saturation amount
+	float Sync; //Hard-sync frequency multiplier
+	float Chop; //Per-period waveform damping
+	float Attack; //Amplitude attack time (fade-in)
+	float Decay; //Amplitude decay time (fade-out to sustain value)
+	float Sustain; //Amplitude sustain value
+	float Release; //Amplitude release time (fade out to silence)
+};
+
+//Renders the sound for each note or something
+class SynchroNote
+{
+	MM_OPERATORS
+public:
+	//Constructor
+	SynchroNote(NotePlayHandle * nph);
+	//Renders a single sample of audio
+	void nextStringSample(sampleFrame &outputSample, sample_rate_t sample_rate,
+		const float & modulationStrength, const float & modulationAmount, const float & harmonics,
+		const SynchroOscillatorSettings & carrier, const SynchroOscillatorSettings & modulator);
+private:
+	NotePlayHandle * nph;
+	float carrierSampleIndex = 0; //The index (or phase) of the carrier oscillator
+	float modulatorSampleIndex = 0; //The index (or phase) of the modulator oscillator
+};
+
+//Synth GUI
+class SynchroSynthView : public InstrumentView
+{
+	Q_OBJECT
+public:
+	SynchroSynthView(Instrument * instrument, QWidget * parent);
+	QSize sizeHint() const override { return QSize(448, 250); } //~3px wider than the artwork
+protected slots:
+private:
+	virtual void modelChanged();
```

Comment:
When overriding methods, omit `virtual` and add `override` (`void modelChanged() override;`). There are a few other instances to look at in this file.